File: Practicals/Projects/Hospital Management System/Frontend/src/app/medicine.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Medicine } from './medicine';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class MedicineService 
{

  private baseURL = "http://localhost:8080/api/v1/medicines";

  constructor(private httpClient: HttpClient) { }

  getMedicineList(): Observable<Medicine[]> 
  {
    return this.httpClient.get<Medicine[]>(`${this.baseURL}`);
  }

createMedicine(medicine: Medicine): Observable<Medicine>
{
   
  return this.httpClient.post<Medicine>(`${this.baseURL}`, medicine);
}

getMedicineById(id:number): Observable<Medicine>
{
  return this.httpClient.get<Medicine>(`${this.baseURL}/${id}`);
}

updateMedicine(id:number, medicine:Medicine): Observable<Object>
{
  
  return this.httpClient.put(`${this.baseURL}/${id}`,medicine);
}

deleteMedicine(id:number): Observable<any>
{
  return this.httpClient.delete(`${this.baseURL}/${id}`,{ responseType: 'text' });
}

getTotalMedicines(): Observable<any>
{
  return this.httpClient.get(`${this.baseURL}/total`);
}
}

File: Practicals/Projects/Hospital Management System/Backend/src/main/java/com/HospitalManagement/demo/Service/MedicineService.java
package com.HospitalManagement.demo.Service;

import java.util.List;

import com.HospitalManagement.demo.Model.Medicine;

public interface MedicineService 
{
	Medicine addMedicine(Medicine medicine);
	
	List<Medicine> getAllMedicines();
	
	Medicine updateMedicine(Medicine medicine, Long id);
	
	Medicine getMedicineById(Long id);
	
	void deleteMedicine(Long id);
	
	long getTotalMedicines();

}
